mutalyzer/grammar.py

```diff
@@ -83,9 +83,19 @@ class Grammar():
     # BNF: GeneName -> ([a-Z] | [0-9] | `-')+
     GeneName = Word(unicode(alphanums) + '-', min=1)
 
-    # BNF: GeneSymbol -> `(' Name (TransVar | ProtIso)? `)'
-    GeneSymbol = Suppress('(') + Group(GeneName('GeneSymbol') + \
-                 Optional(TransVar ^ ProtIso))('Gene') + Suppress(')')
+    # BNF: GeneProductID -> GeneName (TransVar | ProtIso)
+    GeneProductID = Group(GeneName('GeneSymbol') + \
+                          Optional(TransVar ^ ProtIso))('Gene')
+
+    # BNF: AccNoStem -> ([a-Z] Number `_')+
+    AccNoStem = NotAny('LRG_') + Combine(Word(unicode(alphas) + '_') + Number)
+
+    # BNF: AccNoFull -> AccNoStem `.' Number
+    AccNoFull = AccNoStem + Suppress('.') + Number
+
+    # BNF: GeneSymbol -> `(' (GeneProductID | AccNoFull) `)'
+    GeneSymbol = Suppress('(') + (GeneProductID ^ AccNoFull('AccNoTransVar')) \
+                 + Suppress(')')
```

Comment:
On the other hand, we could also be completely agnostic of the accession format here. It might contain a dot, it might not, but we always match it from start to end with the annotated "transcript id".

Only restriction would then be to disambiguate it from the `GENE_v001` or `GENE_p001` option.

What do you think?